frontend/src/components/dashboard/create/util.ts

```diff
@@ -0,0 +1,21 @@
+import { sendCampaign } from 'services/campaign.service'
+import { Campaign, ChannelType, Status } from 'classes'
+import { GA_USER_EVENTS, sendUserEvent } from 'services/ga.service'
+
+export const confirmSendCampaign = async ({
+  campaignId,
+  sendRate,
+  channelType,
+  updateCampaign,
+}: {
+  campaignId: number
+  sendRate: number
+  channelType: ChannelType
+  updateCampaign: (campaign: Partial<Campaign>) => void
+}) => {
+  await sendCampaign(campaignId, sendRate)
+  if (sendRate) {
+    sendUserEvent(GA_USER_EVENTS.USE_SEND_RATE, channelType)
+  }
+  updateCampaign({ status: Status.Sending, sentAt: new Date().toISOString() })
```

Comment:
I'm not sure if we should instead fetch the `sent_at` value from the backend instead. In practice, I don't think the disparity between the 2 values will be significant, since the `sent_at` value is set immediately in the backend anyway.

Reply:
Gave some thought to this, would prefer to refetch the campaign details and update `campaign` in `CampaignContext`. Would prefer not to sacrifice de-sync between local and backend state to save one network call. Was thinking we can make call to backend to refetch `campaign` once status changed to `Status.Sent`. 

Maybe something like that?

```ts
useEffect(() => {
  let timeoutId: NodeJS.Timeout

  async function poll() {
    const { status } = await refreshCampaignStats(id)

    if (status !== Status.Sent) {
      timeoutId = setTimeout(poll, 2000)
    } else if (status === Status.Sent) {
      const updated = await getCampaignDetails(campaign.id)
      setCampaign(updated)
    }
  }

  if (stats.status !== Status.Sent) poll()
  return () => {
    timeoutId && clearTimeout(timeoutId)
  }
}, [id, stats.status, campaign, setCampaign])
```

To extend this further, was refactoring the stats fetching/polling logic in `(Email|SMS|Telegram)Detail` component into a custom hook.

Reply:
That sounds good. I was also a little irked by the implication of de-sync. I'll explore your solution as well as the refactor.

Reply:
Also, I also have some questions regarding possible de-sync of `Campaign` state between the frontend and backend at various steps during campaign creation. I'll look to message you during the weekdays about it :D